### PR TITLE
Support @import #1

### PR DESCRIPTION
**Pull request by erel**
Basic solution for support imports in stylus files

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -18,48 +18,70 @@ StylusFilter.prototype.constructor = StylusFilter;
 StylusFilter.prototype.extensions = ['styl'];
 StylusFilter.prototype.targetExtension = 'css';
 
-StylusFilter.prototype.processString = function (str) {
```
Comment (shinnn) [inline index.js]: Do not remove the speces after `function`s.

File: index.js [hunk 2]
```diff
@@ -18,48 +18,70 @@ StylusFilter.prototype.constructor = StylusFilter;
...
+		src = this.inputTree.inputTree.inputTree + '/' + this.inputTree.srcDir + '/' + src;
+
+		if (!opts.import) {
+			opts.import = [];
+		}
+
+		opts.import.push(src);
+
+		str = str.replace(importRegex, '');
+	}
```
Comment (sindresorhus) [inline index.js]: Can you elaborate on what this code does?

File: index.js [hunk 3]
```diff
@@ -18,48 +18,70 @@ StylusFilter.prototype.constructor = StylusFilter;
...
+	while (importRegex.test(str)) {
+		var imp = importRegex.exec(str);
+		var sourceReg = new RegExp(/['"].*['"]/);
+		var extReg = new RegExp(/.styl/);
+		var src = sourceReg.exec(imp[0])[0].replace(/"/g, '').replace(/'/g, '');
+		if (!extReg.test(src)) {
+			src = src + '.styl';
+		}
+
+		src = this.inputTree.inputTree.inputTree + '/' + this.inputTree.srcDir + '/' + src;
```
Comment (sindresorhus) [inline index.js]: `this.inputTree.inputTree.inputTree`?

Comment (erel) [inline index.js]: If you'll remove this line of code you'll get 'failed to locate @import file...'
That's the way I found for building the full path of the file (I'm using [broccoli-funnel]https://github.com/broccolijs/broccoli-funnel)
If you have more elegant way to get the path of the import file I'll be happy to learn.
Thanks.

Comment (sindresorhus) [inline index.js]: I'm not asking you to remove it. I'm asking what it does and why. Some comments on would be useful.

File: index.js [hunk 4]
```diff
@@ -20,6 +20,36 @@ StylusFilter.prototype.targetExtension = 'css';
 
 StylusFilter.prototype.processString = function (str) {
 	var opts = this.options;
+
+	var importRegex = new RegExp(/@import ['"].*['"]/);
```
Comment (arthurvr) [inline index.js]: Should this be it's own module, just like we have https://github.com/kevva/import-regex?

File: index.js (hunk 2 above)
Comment (arthurvr) [inline index.js]: @erel :arrow_up:

File: index.js [hunk 5]
```diff
@@ -20,6 +20,36 @@ StylusFilter.prototype.targetExtension = 'css';
...
+	while (importRegex.test(str)) {
+		var imp = importRegex.exec(str);
+		var sourceReg = new RegExp(/['"].*['"]/);
+		var extReg = new RegExp(/.styl/);
```
Comment (arthurvr) [inline index.js]: The usage of `.` here is incorrect. If you want to match `.styl`, the dot must be escaped.

File: index.js [hunk 6]
```diff
@@ -20,6 +20,36 @@ StylusFilter.prototype.targetExtension = 'css';
...
+
+		// get import file src and remove quotes
+		var src = sourceReg.exec(imp[0])[0].replace(/"/g, '').replace(/'/g, '');
+
+		// add '.styl' to filename if not exist
+		if (!extReg.test(src)) {
```
Comment (arthurvr) [inline index.js]: Why is this done using a regex? Just check the extension using https://github.com/kevva/ext-name?

**Comment by sindresorhus** [comment]
I can see what the code is doing, but I'm still not sure what the intent of this PR is. What problem is it solving? Why?

**Comment by erel** [comment]
I tried to cover stylus file with @import "reset.css" and the module failed to convert it and this code fix it.

**Comment by sindresorhus** [comment]
This is just a plugin. Handling imports should be handled by the stylus lib. I think you're trying to fix the problem in the wrong place.

**Pull request closed without merging**